test/e2e/metrics_test.go

```diff
@@ -0,0 +1,43 @@
+package e2e
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	core "k8s.io/api/core/v1"
+)
+
+// testMetrics deploys Linux pod and tests that it can communicate with Windows node's metrics port
+func testMetrics(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+
+	// Need at least one Windows node to run these tests, throwing error if this condition is not met
+	require.GreaterOrEqualf(t, len(gc.nodes), 1, "insufficient number of Windows nodes to run tests across"+
+		" nodes, Minimum node count: 1, Current node count: %d", len(gc.nodes))
+
+	for _, winNode := range gc.nodes {
+		t.Run(winNode.Name, func(t *testing.T) {
+			// Get the node internal IP so we can curl it
+			var winNodeInternalIP string
+			for _, address := range winNode.Status.Addresses {
+				if address.Type == core.NodeInternalIP {
+					winNodeInternalIP = address.Address
+				}
+			}
+
+			// This will curl the windows server. curl must be present in the container image.
+			linuxCurlerCommand := []string{"bash", "-c", "curl http://" + winNodeInternalIP + ":9182/metrics"}
+			linuxCurlerJob, err := testCtx.createLinuxJob("linux-curler-"+strings.ToLower(winNode.Status.NodeInfo.MachineID),
```

Comment:
Should this be accessible from linux workloads or worker node directly?

Reply:
I think it should be accessible from both, linux workloads and worker nodes. Please correct me if I am wrong, as I am hitting CI issues with the current test setup. Would it be better to test from worker node directly?

Reply:
We need to mimic whatever entity that will be accessing this endpoint. I am not completely sure of where or how that entity runs. Something to look into.

Reply:
I strongly assume this runs in a pod somewhere - there is no host service that collects the metrics afaik